http-api/src/lib.rs

```diff
@@ -230,6 +244,32 @@ async fn blob_handler(
     Ok(warp::reply::json(&blob))
 }
 
+async fn commit_root_handler(
+    ctx: Context,
+    project: Urn,
+    revision: String,
+) -> Result<impl Reply, Rejection> {
+    let reference = Reference::head(
+        Namespace::from(project.clone()),
+        None,
+        revision.clone().try_into().unwrap(),
+    );
+    let revision = Revision::<PeerId>::Sha {
+        sha: revision.as_str().try_into().unwrap(),
```

Comment:
I'm also wondering what's going on here, since we're setting a `sha` to some branch name?

Reply:
hmm the `revision` is the argument passed into the function.
We could eventually rename it to `head` or `commit` to make it clear that it's the head from where the commit tree gets built.

Reply:
The problem is not the name, it's that we're passing it to a `Revision::Sha`, when I'd expect that it only handles SHA-1s. There should be a `Revision::Branch` or something for branches? And if we want to support commit histories from any type of revision, ie. tags, branches and commits, we need to create the right kind of `Revision` object.

Reply:
> Can we propagate the error here and above, instead of unwrap?

In the former, I would leave unwrap since `warp::reject::Reject` is not implemented for `radicle_daemon::radicle_git_ext::name::Error`

For the latter one I was able to propagate errors.

Reply:
Okay nevermind, was able to map errors to `Error::NotFound` of `error.rs` which I think is the most reasonable when not able to find a commit..
Did also some better error handling on the other routes.